cpp/src/arrow/csv/writer.cc

```diff
@@ -548,6 +548,18 @@ Status WriteCSV(const RecordBatch& batch, const WriteOptions& options,
   return writer->Close();
 }
 
+Status WriteCSV(const std::shared_ptr<RecordBatchReader>& reader,
+                const WriteOptions& options, arrow::io::OutputStream* output) {
+  ASSIGN_OR_RAISE(auto writer, MakeCSVWriter(output, reader->schema(), options));
```

Comment:
nit: I realize it's inconsistent above, but can we stick with either of ARROW_ASSIGN_OR_RAISE or ASSIGN_OR_RAISE here?